### ISSUE TITLE
Fix compatibility with stylelint-processor-styled-components

### ISSUE THREAD

**Issue by yoannmoinet** [stylelint/stylelint#3252]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Might be a bug with the latest 9.2.0.

> Which rule, if any, is this issue related to?

A specific processor, [stylelint-processor-styled-components](https://github.com/styled-components/stylelint-processor-styled-components) (the only one I'm using) isn't working after 9.1.3.

> What CSS is needed to reproduce this issue?

It will have to be a css using a processor, in my case a simple js file using [styled-component](https://github.com/styled-components/styled-components)  will do : 

```javascript
// test.js
import styled from 'styled-component';

export default styled.div`
.test {
background:black;
}
`;
```

> What stylelint configuration is needed to reproduce this issue?

Any config as long as it's using a processor.

```json
// config.json
{
  "processors": ["stylelint-processor-styled-components"],
  "extends": [
    "stylelint-config-standard",
    "stylelint-config-styled-components"
  ]
}
```

> Which version of stylelint are you using?

`9.2.0` not working
`9.1.3` working

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint --config config.json test.js`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's using styled-components.

> What did you expect to happen?

Having errors written in the terminal.

> What actually happened (e.g. what warnings or errors you are getting)?

No output at all.

I tried the exact same configuration with `9.1.3` and I got errors in the terminal.
Nothing with `9.2.0`.

I'm not sure if it's coming from `stylelint` or `stylelint-processor-styled-components`] so I opened an error in both with as much info I can get. 
styled-components/stylelint-processor-styled-components#180
Hope that will help.

**Comment by ntwb** [comment]
> I believe we need to resolve what the roles of `postcss-html` and processors are. It looks like [changes are happening](https://github.com/stylelint/stylelint/issues/3248#issuecomment-378213074) in `postcss-html` that break existing processors.

I was looking at `postcss-html` the other day and initailly started considering if it should be split up, I was thinking `postcss-vue`, `postcss-html`, `postcss-php`, `postcss-markdown` though I abandoned that thought because `postcss-html` is _"HTML (and HTML-like)"_ and there would be considerable overlap between `postcss-html` and `postcss-php`, possibly also `postcss-vue`. Pulling out `postcss-markdown` I think would have little overlap, see below.

> Does `postcss-html` seek [to replace processors](https://github.com/stylelint/stylelint/issues/3218#issuecomment-374080935)?

Pulling `markdown` out of `postcss-html` into `postcss-markdown` with its Remark dependancies would at least help out with this by reducing the size of `postcss-html`. 

> I believe the disadvantage of processors is that they don't currently support autofixing (and it may not be technically possible to make them work with it).

`postcss-markdown` could be a canditate to test the above with, though with the Remark universe it might just be possible with the myriad of available Remark plugins 😏

**Comment by jeddy3** [comment]
> "HTML (and HTML-like)"

Maybe it should just work on `<style>` tags and `style` attributes? It seems like a common use case (Vue, Web components and plain ol' vanilla HTML pages) that we should bundle within core.

> Pulling `markdown` out of `postcss-html` into `postcss-markdown` with its Remark dependancies would at least help out with this by reducing the size of `postcss-html`.

Agreed. It doesn't seem like such a common use. Perhaps also pull out the template literal parsing too into a separate package? 

> `postcss-markdown` could be a candidate to test the above with

Could be. I'm still not 100% sure what using a syntax over a processor offers? Are we sure that autofix can only be implemented via a syntax?

**Comment by stnwk** [comment]
Is there any quickfix available for this issue?

**Comment by gucong3000** [comment]
> Is there any quickfix available for this issue?

just run `npm i --save stylelint/stylelint#issue-3248`

**Comment by shth** [comment]
@gucong3000 I tried your quickfix and it's working fine for styled-components. But now linting for .scss files are not working (no error output). Is that expected?

**Comment by evilebottnawi** [comment]
@gucong3000 looks like we should documentation this, or maybe do this feature built-in :thinking:

**Comment by jeddy3** [comment]
The fix for the regression was released in `9.2.1`.
